### PR TITLE
Enable default performance hint as tput in AUTO

### PR DESCRIPTION
**Pull request by yangwang201911**
Enable default performance hint as tput in AUTO

### Details:
 - Enable hint to tput if no property is specified for both AUTO device and target device.

### Tickets:
 - CVS-84501

### CODE REVIEW

**Review by peterchen-intel — APPROVED** [review]

**Review by wangleis — APPROVED** [review]

File: src/plugins/auto/plugin.cpp [hunk 1]
```diff
@@ -106,8 +106,12 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::ParseMetaDevices(cons
         if (!deviceIDLocal.empty()) {
             tconfig[PluginConfigParams::KEY_DEVICE_ID] = deviceIDLocal;
         }
-
-        return GetCore()->GetSupportedConfig(deviceName, tconfig);
+        auto deviceConfig = GetCore()->GetSupportedConfig(deviceName, tconfig);
+        if (deviceConfig.find(PluginConfigParams::KEY_PERFORMANCE_HINT) == deviceConfig.end() && tconfig.find(deviceName) == tconfig.end()) {
+            // setting tput as the default performance mode if no hints setting for AUTO plugin and no properties specified for target device.
+            deviceConfig[PluginConfigParams::KEY_PERFORMANCE_HINT] = PluginConfigParams::THROUGHPUT;
```
Comment (songbell) [inline src/plugins/auto/plugin.cpp:112]: I found that benchmark_app -d MULTI:GPU -hint latency  will trigger tput as well, is this as expected?
@yangwang201911

Comment (songbell) [reply src/plugins/auto/plugin.cpp:112]: @peterchen-intel @yangwang201911 can we test on DG1 to see if this functions well?

Comment (yangwang201911) [reply src/plugins/auto/plugin.cpp:112]: Below steps could reproduce this issue. The root cause for this exception is ` deviceConfig = GetCore()->GetSupportedConfig(deviceName, tconfig); ` didn't retrieve the config setting for CPU plugin. Function `GetSupportedConfig(deviceName, tconfig);` just merge the config passed through `compiled_model` (including the secondary level properties) if this config is one of the supported keys by CPU plugin. We need to update the function `GetSupportedConfig();` to merge the configs set via function `set_property()` before calling `compiled_model()` into the final configs list for "CPU" plugin. Is this ok to fix this issue? @peterchen-intel @wangleis 

`
core.get_property("CPU", ov::supported_properties);
core.get_property("MULTI", ov::supported_properties);
core.set_property("CPU", {ov::hint::performance_mode(ov::hint::PerformanceMode::LATENCY)});
ov::CompiledModel compiled_model = core.compile_model(model, "MULTI:CPU");
`

**Comment by ilya-lavrenov** [comment]
why AUTO is tput oriented? I suppose MULTI should be tput, while for AUTO it's questionable..

**Pull request merged**